### DISCUSSION TITLE
Please, improve insert feature.

### DISCUSSION THREAD

**Discussion by mn1712trungson** [WerWolv/ImHex#1110] [Feedback]
Add an option or shortcut to type text in insert or overwrite mode, like an IDE (VIM, Visual, etc)
or text editor (Word, MSWord, LibreOffice Writer, etc) (the shortcut should be the INS key).

Sometimes, there will be times when we need to insert more text at a specific address, but right now it is quite painful to have
to Right-click > Insert > Select size and remove the empty space.

https://www.youtube.com/watch?v=3xyF_KVeFPk

**Comment by WerWolv** [comment]
ImHex is not a text editor and does not try to be one.
Inserting bytes into binary data almost always results in the structure of the file to be completely messed up, that's why this option is behind a menu option that can't be used accidentally. One of my biggest pet peeves with other hex editors is that they sometimes just randomly resize my file which is basically never what I actually wanted to do.
